filter/filter.go

```diff
@@ -193,15 +198,37 @@ func searchRecordString(term string) Filter {
 		}
 	}
 	searchContainer := Contains(search)
-
+	fieldNameCheck := make(map[int]bool)
 	return func(r *zng.Record) bool {
+		// Memoize the result of a search across the names in the
+		// record columns for each unique record type.
+		id := r.Type.ID()
+		match, ok := fieldNameCheck[id]
```

Comment:
The scanner will only work if the typeIDs in the value buffer match the typeIDs in its context.  If that gets messed up then nothing works, so I don't understand what you mean by unlikely Reset().  If you are envisioning some sort of race during parallel decoding, then that would be a bug.  What am I missing here?

Reply:
Good catch.  This context is different from the flow graph because it's in the scanner..  If we adopted your fix, then unused type pointers would hang in the table during the life of the scan.  That seems ok.

Reply:
Issue #1478 for "another forum" topic.